### PR TITLE
Optimize test case in migrating slots

### PR DESCRIPTION
**Pull request by haorenfsa**
### Status

DEVELOPMENT

## Description

Add test case when the key being queried is migrating.

## Todos

- [x] Unit Tests

**It will be awesome if you cover the negative test cases as well.**

## Notes to reviewers

Review the following things, including but not limit to:

1. Test cases (**This is a MUST**, don't pass the PR if it's missing)
2. Design
3. Code (styles, magic...)

### CODE REVIEW

File: proc/redis/session.go [hunk 1]
```diff
@@ -108,6 +108,10 @@ func (s *session) loopWrite() {
 		req.Wait()
 		// TODO(kirk91): abstract response
 		resp := req.Response()
+		// asking request is added by sam during slot migration, thus resp should be swallowed
+		if req.body.Equal(askingRequest) {
+			continue
+		}
```
Comment (doyoubi) [inline proc/redis/session.go]: A bit weird... The ASKING request created by upstream shouldn't be inside `processingReqs` of session.

Comment (haorenfsa) [reply proc/redis/session.go]: yes! the req will be some kind of `get`,  may be use a filter will be a better solution?

Comment (doyoubi) [reply proc/redis/session.go]: I mean, as session and client(upstream) have their own `processingReqs`, the ASKING request created by upstream should just be inside `client.processingReqs`, not the one in session. 
And I can't find any codes that will pass ASKING reques from `client.processingReqs` to `session.processingReqs`.

Comment (haorenfsa) [reply proc/redis/session.go]: > I mean, as session and client(upstream) have their own `processingReqs`, the ASKING request created by upstream should just be inside `client.processingReqs`, not the one in session. 

yeah, i got it

> And I can't find any codes that will pass ASKING request to from `client.processingReqs` to `session.processingReqs`.

it's done in the hook, the hook will be called when upstream request finished, like below:

```go
	simpleReq.RegisterHook(func(simpleReq *simpleRequest) {
		req.SetResponse(simpleReq.Response())
	})
```

Comment (haorenfsa) [reply proc/redis/session.go]:
```go
		askingReq := newSimpleRequest(newArray(
			*newBulkString(ASKING),
		))
		u.MakeRequestToHost(hostAddr, askingReq)
		u.MakeRequestToHost(hostAddr, req)
```

So I took a closer look seems no bug for the opensource version, I'll try to run the integration test local to make sure. And may be eleme's code version is in different logic.

Comment (haorenfsa) [reply proc/redis/session.go]: The test passed, So we just need to update the test

**Pull request closed without merging**